torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -2262,6 +2311,20 @@ def _load_sharded_state_dict(
         with self.set_state_dict_type(StateDictType.SHARDED_STATE_DICT):
             return self.load_state_dict(state_dict, strict)
 
+    def _params_exec_order_hook(self, named_param: Tuple[str, FlatParameter], *unused: Any) -> None:
+        # In self._fsdp_named_params_exec_order, the parameters are ordered based on
+        # the execution order in the backward pass.
+        self._fsdp_named_params_exec_order.append(named_param)
+
+    def _register_params_exec_order_hook(self) -> None:
```

Comment:
what is the difference from _register_post_backward_hooks()? it should be able to reuse post_backward_hook() to add the '_fsdp_named_params_exec_order'

Reply:
Just moved it to post_backward_hook()